modules/dists/DimensionalDist2D.chpl

```diff
@@ -490,6 +490,30 @@ proc DimensionalDist2D.dsiClone(): _to_unmanaged(this.type) {
   return _to_unmanaged(this);
 }
 
+//== targetLocales, localSubdomain
+
+proc DimensionalDist2D.dsiTargetLocales() const ref return targetLocales;
+proc DimensionalDom.dsiTargetLocales()    const ref return dist.targetLocales;
+proc DimensionalArr.dsiTargetLocales()    const ref return dom.dist.targetLocales;
+
+proc DimensionalDom.dsiHasSingleLocalSubdomain() param return true;
+proc DimensionalArr.dsiHasSingleLocalSubdomain() param return true;
+
+proc DimensionalDom.dsiLocalSubdomain(loc: locale) {
+  import ChplConfig;
+  if ChplConfig.CHPL_COMM != "none" then
+    compilerError("DimensionalDist2D.dsiLocalSubdomain() is currently not implemented for non-local compilation");
+  return whole;
+  // for multilocale, start here:
+  // const (locIds, foundIt) = _CurrentLocaleToLocIDs(dist.targetLocales, loc);
+  // var result: whole.type;
+  // if !foundIt then return empty result;
```

Comment:
Is this code intentionally commented out?

Reply:
Yes. I started working on this method, however timed out, so left these to transcribe where I have gotten so far.